### ISSUE TITLE
azurerm_frontdoor plan crash after upgrade to 2.73.0

### ISSUE THREAD

**Issue by cpressland** [hashicorp/terraform-provider-azurerm#13077]
### Terraform (and AzureRM Provider) Version

```
$ terraform --version
Terraform v1.0.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azurerm v2.73.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/terraform-providers/chef v0.2.0
+ provider registry.terraform.io/terrycain/chef v0.3.2
+ provider registry.terraform.io/terrycain/commandpersistence v1.1.0
```

### Affected Resource(s)

* `azurerm_frontdoor`

### Terraform Configuration Files

```
-----BEGIN PGP MESSAGE-----
Version: Keybase OpenPGP v2.1.13
Comment: https://keybase.io/crypto

wcFMA1RCL2EjebqaARAAkSHopQFaDp7gg3rjNAMQ2JcKej+ENfIBaZoK8aCeHjLx
HNgS8jcc9sIGOnc3RW2EFeJ+v4GbgFoSRFB6zeU/rSC9X7O1ftSCpeN48l1Nv+5K
KKC/rwrE0EEiianXphzF2D8b92LIDW7BdxmDy9k4zd5PyFBUjJH0ltH8kBpEDF47
NYkkD5HpX65oAkYMpbftCfDELpK8PVxe2E15nqRMnYYfxnICG8v2/k5xgPyzEzV2
VysxDeQy/r6S6uIdVIoW7r7+AaAIMEAEckjjWu+yXDSZaLb8RdNssDKgWomj0NKx
o2wuruvjjECbzWYmnaTdO57PJx1Hb1fJJHqcF3Vs9eh26xqG8kJU0mVbBfA4GEBh
tg2ACNR/WHczT2AzrjuK2tclCjzNqCtybcHMmOJm/qlq8OaqS64AGm0dKUAIuDDu
FAIAP8bkKw0dvCDvKtA51Cy/OmwpUvEBTvc1ZXek5dDedZiIjzwiU7k0PzdZ/7Fl
ryJPMorRfWrXc5GqNV8FpHwOdg8+K1MFscqgxycPdsY8vmUXAnDqrOGx64COtdFd
TN/kZ0sbELMstFKAFfRngITa5GWYapSrYdItSyxSboU7WkqKi4Qf2hjK7uWjginw
THu7tiXtIxUVvNNhdNa+X3NpzFGne6LGoEEDzuszClOSNufQEuasawpDT/ClCS7S
x9AB5gPWcfq89tnO1FoCFsNiUpm6qdEHZ+bcqWKXQD5kvakEGjngtpU1ve2LuEcD
jUmo5+8LAZ36xV+/Vd7M5GF03ZQwuuRBfEh2EZ7G/AsIT6E7Mg1T7ownAdro6rGR
4mtO5XUIJn9vpwkiID0MP7udmbK+hXuwu8KkKllbJvesAsFV5WY0loTd2rLQygBN
Iogj+9GVDGxp5vgNlbKzv/ik3Rf1YAsua6WJ9i0VBRIiwTB7Js5gh2vGcEePnRUA
07MRKR1WXr4ngPunJH+kPHY7vhBSTravcR990SbagdPYYyj32645cijjcdhFD678
QTPOmWrFzYIaIvWggkwo7hi8Mf/7elW5YI94MgsYjUJ+IQBmnXausi7+Vl+LnsgY
0C9Dfk8JZQVLlCteFRxhvZ9ngl5YsI+9NVUkRsVlQnivVjPit9jn99W6I/29R1uX
eMxjirwxEIOX/4krHbDNXhRSvCbvatIXCGHXALppYz5ruCi1Y+l5HtdwRRcGhZS7
lENdtwcvKeV7Hd5ofVWu+9G0zngSRSgf2Uk0G/5dbW1+dfEFgwrTB3WQrEBhsnYP
JEdJlNneOxAE3/ksC7jvV9Ku+WBMV36xCTlpxBhXZ7G5SWJ9SUsu5LIaR/SnYRE0
VvnprZrfKij/uRY4sN6JATENL9tKGQEY7QKYG38k9kOSybYAkSemEOlf1CngSLZa
ckmwsbTTMQNeHMIq34VMaLvyukoJd0XpcEMiAA6eiPkJT/pXY1VYrOumQewgm+ea
MxvlcTkR8binEwsKFLe3sudrXfVpgd9l8f2D+pnoOVwltMwYT2x0VtHr2ppF7GO7
Yw0UIKvAePG6Hi7TtRp/bLbNFV0H1kOlWfy4IYI7fTacBxyRhdn8gTV2SEHOXm4k
fxEljqK6l5jxWIxPWTJkgVVULauGPentX5XRoKK30Qr6RQJE6zNCM6xwBJPWdNXT
qKfq4JHywIx0rCk6KaGjdnrd75ohqPRdoaugMyRUIJXeUPIVesj6VA306ELn29lV
bWzJbncuMxAyj5n9BQhnDVoy0nYRwiyBu3AOz85h7uRHX5YsP/xl3E7x0l99M1Sr
Pi/Vp8dk8LtE+QLMndekruImEDt/FfAUiLMXWW1T3rEo9OX6ff9oBeXRdl2ocaUo
XDB9Tl4C4z3lF7UK+dknf1Gli00rrzMFaMkpyjiwTnMAhKjYE/wVJD1feBrEbYrr
4F0zC8r1MB+2fjQLd9kdatjEzhaRjdWuVoT3KgHcvHH20xtSAXM3eL/69bVXXbKj
/NmB+de7H6SDu2YIQm3yD0L9F/n/JzrUYxWdaz8LTchpnTXtHeTFOk8HY4/T9SMk
lmEyUApAhBjLbNFfTx4WNKlya3AhLIuMrl7YiwRc7/kKp92mRtwsfARLI3wLwx/9
rO78RbOYXnUNXbny93XyZ+Na9Ej9H7BpsTCDIBct+imkNcxqwxf+RYLQAMhDPFhL
ipSICs8kRc6PTeC9AlRkFE2KzrLp28kT9ihrlkD5jZrvssHma8nC/JEItbdh+g7h
4TLCSI+i05j08EqILW3TC3Y+BKCDFiGiO/S0CHE6IIU7wuoXEwi9wMlzVUw4xiWa
zLqkR4leytMCKb3hujC9rAvgxpV+g7P3a9i0GohgDgAT2KEge3OcCc7JAYl9X8Xh
hlXv6Uini+HvjJPPwSRfHSSLKnCekpVXyhiVCSiEZslZkUsi/e4Xg3692M2nVC6Q
01WqZyrGfl37YUriTMe4e1HLFijSc/rI12J9J2DpTxE4hSJeEWk8D1x7SqEktr1M
0kwrdN/zNyCTgINvSJ6rTEq9Q6CHaY11DoEdafYPjtpJRLkxjSLPCRgLuGDG+eKq
q+OtZooYIHfMquP0sMLsJ53tktq86l7wj7hIeMLUfxypc/xEXuz+0EtfEvS6PuXL
b9qf7uNuPsieYbM24FuE7Ao37wSJrUkyae8nT+10izdjNFM7QHOpdewpRyskGVaE
5hRkOwEey+tS2QOa/X56pPj+31GT1d3i/GPgTdESRAiz0WeqRpv7jHKPPKKEHLiK
n4t2f4EbtXaWjqV1mOJvlqHF7cw1dNOqkW/MGBgT5eao7Hh5z6Z1QqA7MdT7h+qt
MXSUOHjx/witsWLig3dMB/5vrh0uJ/uvpaBcrAoggAdFTsr2yb0QobEAb76yPbNC
rf2/Godfke/jKynC5vdHzmr+OJkP4Ez4TevteGLvkfQAq2kIpXc9saXEtfpbJ64Q
Nvid9yb5jfVnxirvqiegXGOlhezzFkmKPC9CoIMgLVA10+1dDXmrRNcKU0+L1IpS
Z7dCxbCUPO+LwaTMX0F1w2VBtiKr//s7qP0nNi82gajgFZdGxdV7T1THpU/uV2L9
EBosiyxjTJpDymIvV+um0gucCkJwH0pEnEVpA8rQfEq7hz42SzyUqGKwb08iaHfe
AeRnB7neyftadmmqeVJvbwMMXpNTE4XoKyyr4Sm6m1huhsLEhnetqBWFF+6DEnDX
zGNRa1AckPl4elAgF36a6/0kWG/h1oEOGNkcy/lgdRbyQISTQB9LKkVTH1xRTjqP
Ehm5ZF57xf8ZauF/YkgtwrjlgIbAPTeyxXxcJ3Vf2AqswyMSqB4Ym5JgeQItvXSB
tZxwzlRD8lPDUcH5U03wunCjCSj+rwXRTooozEYwcYyX0JhDilYYrj9evrXG66V9
jAo5nHqYbwUORuUbyfmkRnrRWHJT2tGsiwZRr4wBxmp90juWhl3BvQSbUaVrXLjf
ww+Ttb7vlPf8X1pp2oOFB/huX7z6LnGOs9m2wM3u7Yzr+x+lnbeUUps+2bAtVNj+
DqR0pIJyiqBQ3+e30SafLbbyo9yhYDSd+XIHzt2gyc1lcO0mtDEnh6Fs0CeTAlkf
ObixiAoq0jUCw/WxdsspeGGsaJMQI+m7svGAwKCV5686fg==
=YjpK
-----END PGP MESSAGE-----

```

### Panic Output

```
╷
│ Error: Request cancelled
│ 
│   with module.uksouth-frontdoor.azurerm_frontdoor.frontdoor,
│   on uksouth/frontdoor/main.tf line 60, in resource "azurerm_frontdoor" "frontdoor":
│   60: resource "azurerm_frontdoor" "frontdoor" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v2.73.0_x5 plugin:

panic: interface conversion: interface {} is []interface {}, not string

goroutine 9392 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.flattenRoutingRuleForwardingConfiguration(0x107e851b8, 0x14002a4a200, 0x1073c7940, 0x14001758138, 0x4, 0x140011c2c60, 0xa8)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:2005 +0xc28
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.flattenSingleFrontDoorRoutingRule(0x140024028a0, 0x14002485940, 0x140024857b0, 0x14002485930, 0x1073c7940, 0x14001758138, 0x1400260a5af, 0x24, 0x1400260a5e3, 0x9, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:1916 +0x638
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.combineRoutingRules(0x14000b081a0, 0xc, 0xd, 0x1073c7940, 0x14001758138, 0x14001256f00, 0xc, 0xc, 0x1400260a5af, 0x24, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:1833 +0x480
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.flattenFrontDoorRoutingRule(0x140024aa240, 0x1073c7940, 0x14001758138, 0x1400260a5af, 0x24, 0x1400260a5e3, 0x9, 0x1400260a614, 0xe, 0x14002c55b90, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:1877 +0x138
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.resourceFrontDoorRead(0x14002609a00, 0x1075725e0, 0x14001d57500, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:717 +0xe68
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x140007a62a0, 0x107eb83c8, 0x140022fed80, 0x14002609a00, 0x1075725e0, 0x14001d57500, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x140007a62a0, 0x107eb83c8, 0x140022fed80, 0x140003ed500, 0x1075725e0, 0x14001d57500, 0x1400062a820, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:624 +0x160
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x14000652f78, 0x107eb83c8, 0x140022fed80, 0x140022fedc0, 0x140022fed80, 0x102ee834c, 0x10789dae0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:575 +0x380
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0x140016ea8c0, 0x107eb8470, 0x140022fed80, 0x1400262d200, 0x140016ea8c0, 0x102e52240, 0x1078d3b40)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:298 +0xe4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x107a5ef80, 0x140016ea8c0, 0x107eb8470, 0x140011a9290, 0x1400262d1a0, 0x0, 0x107eb8470, 0x140011a9290, 0x14002e32000, 0x83d9)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000220a80, 0x107f06278, 0x14000602480, 0x1400222a7e0, 0x14000438990, 0x10b45c010, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x14000220a80, 0x107f06278, 0x14000602480, 0x1400222a7e0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x140010e86f0, 0x14000220a80, 0x107f06278, 0x14000602480, 0x1400222a7e0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1f8

Error: The terraform-provider-azurerm_v2.73.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

Nothing, this is happening during a plan where nothing has changed other than a provider upgrade.

### Actual Behaviour

Crash!

### Steps to Reproduce

Run terraform plan on equivalently complex Azure Front Door instance.

**Comment by EpicWink** [comment]
An example Terraform config extract to make up for the one missing in the OP

```terraform
resource "azurerm_frontdoor" "test" {
  name = "test"
  resource_group_name = azurerm_resource_group.rg.name
  enforce_backend_pools_certificate_name_check = false
  frontend_endpoint {
    name = "test-endpoint"
    host_name = "test.azurefd.net"
  }
  backend_pool {
    name = "test-backend-pool"
    health_probe_name = "test-health-probe"
    load_balancing_name = "test-load-balance"
    backend {
      address = azurerm_app_service.test-01.default_site_hostname
      host_header = azurerm_app_service.test-02.default_site_hostname
      http_port = 80
      https_port = 443
    }
    backend {
      address = azurerm_app_service.test-01.default_site_hostname
      host_header = azurerm_app_service.test-01.default_site_hostname
      http_port = 80
      https_port = 443
    }
  }
  backend_pool_health_probe {
    name = "test-health-probe"
  }
  backend_pool_load_balancing {
    name = "test-load-balance"
  }
  routing_rule {
    name = "test-route"
    frontend_endpoints = ["test-endpoint"]
    accepted_protocols = ["Http", "Https"]
    patterns_to_match = ["/*"]
    forwarding_configuration {
      backend_pool_name = "test-backend-pool"
    }
  }
}
```

**Comment by matthawley** [comment]
I'm also experiencing this, reverting to 2.72.0 is fine.

**Comment by tombuildsstuff** [comment]
Fixed in #13338 - thanks for the cross-link @EpicWink
